src/mixins/object_interactivity.mixin.js

```diff
@@ -88,24 +88,32 @@
           x, y;
 
       for (var control in coords) {
+        // handle custom control corner sizes
+        var controlObject = fabric.Object.prototype.controls[control];
+        var cornerHyp = controlObject.cornerSize
```

Comment:
what about extending the controls to return their own corner shape? is a bit more complicated but would solve the problem of both different sized controls and corners that are rectangular.

controlObject.cornerSizeX and cornerSizeY and would also change the math down to something i can't calculate now on the top of my head. But definitely doable